### DISCUSSION TITLE
Support for file include, fragment, merge or embed?

### DISCUSSION THREAD

**Discussion by SuperOfficeDevNet** [remarkjs/remark#1048] [Q&A] [answered]
I searched the repos pages and discussions, as well as many of the plugin pages, but didn't find a lead as so whether there is existing support for this, Question is: Does remark have a directive, or other means to support including content, or fragments of markdown, from other files? Or does this require a custom directive?

Example markdown:
```markdown
# Heading

Introduction to the page.

## Prerequisites

:::include(./common/preqs.md)

## More

Details about this page.
```

Best regards.

**Comment by wooorm** [comment, marked as answer]
> remark have a directive

https://github.com/remarkjs/remark-directive

> support including content, or fragments of markdown, from other files

There are several ways to do this. I’m sure there are plugins out there. Depends on what you want though.

https://github.com/search?q=remark+include
https://github.com/search?q=remark+import
https://github.com/search?q=remark+embed

> Or does this require a custom directive?

If none of the plugins in the linked search results match what you want, a new plugin (based on `remark-directive`) would do the trick!

**Reply by slavafomin** [reply]
Hello, just found this discussion. I've had [the same question](https://github.com/orgs/remarkjs/discussions/1101). Sadly, all the existing plugins are severely outdated and can't be used with the latest version of remark. I wonder why though, isn't it a pretty basic feature?

**Reply by ChristianMurphy** [reply]
> I wonder why though, isn't it a pretty basic feature?

Not really.
None of HTML, Markdown, or AsciiDoc have this.
They're static document languages not template languages.

If you're interested you could add it though.
There's good info on directives here: https://github.com/remarkjs/remark-directive
And on how to work with AST and create a plugin here: https://unifiedjs.com/learn/
